sqld/src/lib.rs

```diff
@@ -213,9 +215,17 @@ async fn start_replica(
     let enable_bottomless = config.enable_bottomless_replication;
     #[cfg(not(feature = "bottomless"))]
     let enable_bottomless = false;
+    let hook = ReplicationLoggerHook::new(logger.clone());
+
+    // load dump is necessary
+    let dump_loader = DumpLoader::new(config.db_path.clone(), hook.clone()).await?;
+    if let Some(ref path) = config.load_dump_path {
+        dump_loader.load_dump(path.into()).await?;
+    }
```

Comment:
Why only on `start_primary`?

Reply:
If a replica load a dump, then there is no guarantee it's state is consistent with the primary. Hence, loading on a replica is forbidden